rules/okta/defense_evasion_suspicious_okta_user_password_reset_or_unlock_attempts.toml

```diff
@@ -82,6 +82,6 @@ name = "Initial Access"
 reference = "https://attack.mitre.org/tactics/TA0001/"
 
 [rule.threshold]
-field = ["okta.actor.id"]
+field = ["okta.actor.alternate_id"]
```

Comment:
I've seen this change between `okta.target.alternate_id` and `okta.actor.alternate_id`. Does anyone have an environment up and running to test this out and post the redacted event?

Reply:
So it depends on the event type:

For password resets/unlocks, both ```okta.target.alternate_id``` and ```okta.actor.alternate_id``` exist.

Here are a snipped few examples:

<details>
<summary>event.action: "system.sms.send_account_unlock_message"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```
- ```okta.target.alternate_id``` contains both ```<snipped (real user)>``` and ```<snipped (real user's phone number)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "system.sms.send_account_unlock_message",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>": "<snipped>"
      },
      "event_type" : "system.sms.send_account_unlock_message",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Send self-service account unlock SMS message",
      "client" : {
        "<snipped>": "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>": "<snipped>"
      },
      "target" : [
        {
          "id" : "<snipped (real user)>",
          "type" : "User",
          "display_name" : "<snipped (real user)>",
          "alternate_id" : "<snipped (real user)>"
        },
        {
          "id" : "<snipped (real user's phone number)>",
          "type" : "MobilePhone",
          "display_name" : "<snipped (real user's phone number)>",
          "alternate_id" : "<snipped (real user's phone number)>"
        }
      ]
    }
  }
}
```
</details>

<details>
<summary>event.action: "system.sms.send_password_reset_message"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```, with the exception of 1 event which has the value of ```<snipped (real user)>```
- ```okta.target.alternate_id``` contains both ```<snipped (real user)>``` and ```<snipped (real user's phone number)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "system.sms.send_password_reset_message",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>" : "<snipped>"
      },
      "event_type" : "system.sms.send_password_reset_message",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Send self-service password reset SMS message",
      "client" : {
        "<snipped>" : "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>" : "<snipped>"
      },
      "target" : [
        {
          "id" : "<snipped (real user)>",
          "type" : "User",
          "display_name" : "<snipped (real user)>",
          "alternate_id" : "<snipped (real user)>"
        },
        {
          "id" : "<snipped (real user's phone number)>",
          "type" : "MobilePhone",
          "display_name" : "<snipped (real user's phone number)>",
          "alternate_id" : "<snipped (real user's phone number)>"
        }
      ]
    }
  }
}
```
</details>

<details>
<summary>event.action: "system.voice.send_account_unlock_call"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```
- ```okta.target.alternate_id``` contains both ```<snipped (real user)>``` and ```<snipped (real user's phone number)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "system.voice.send_account_unlock_call",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>" : "<snipped>"
      },
      "event_type" : "system.voice.send_account_unlock_call",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Send self-service account unlock call",
      "client" : {
        "<snipped>" : "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>" : "<snipped>"
      },
      "target" : [
        {
          "id" : "<snipped (real user)>",
          "type" : "User",
          "display_name" : "<snipped (real user)>",
          "alternate_id" : "<snipped (real user)>"
        },
        {
          "id" : "<snipped (real user's phone number)>",
          "type" : "MobilePhone",
          "display_name" : "<snipped (real user's phone number)>",
          "alternate_id" : "<snipped (real user's phone number)>"
        }
      ]
    }
  }
}
```
</details>

<details>
<summary>event.action: "system.voice.send_password_reset_call"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```
- ```okta.target.alternate_id``` contains both ```<snipped (real user)>``` and ```<snipped (real user's phone number)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "system.voice.send_password_reset_call",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>" : "<snipped>"
      },
      "event_type" : "system.voice.send_password_reset_call",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Send self-service password reset call",
      "client" : {
        "<snipped>" : "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>" : "<snipped>"
      },
      "target" : [
        {
          "id" : "<snipped (real user)>",
          "type" : "User",
          "display_name" : "<snipped (real user)>",
          "alternate_id" : "<snipped (real user)>"
        },
        {
          "id" : "<snipped (real user's phone number)>",
          "type" : "MobilePhone",
          "display_name" : "<snipped (real user's phone number)>",
          "alternate_id" : "<snipped (real user's phone number)>"
        }
      ]
    }
  }
}
```
</details>

<details>
<summary>event.action: "user.account.unlock_token"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```
- ```okta.target.alternate_id``` contains ```<snipped (real user)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "user.account.unlock_token",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>" : "<snipped>"
      },
      "event_type" : "user.account.unlock_token",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Issued recovery token for self-service account unlock",
      "client" : {
        "<snipped>" : "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS",
        "reason" : "Issued recovery token"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>" : "<snipped>"
      },
      "target" : [
        {
          "type" : "User",
          "alternate_id" : "<snipped (real user)>"
        }
      ]
    }
  }
}
```
</details>

Based off the updated wording of the PR, I believe it is correct that ```okta.actor.alternate_id``` is being used here, as it's looking for a high number of resets/unlocks across the environment(s).


Note: I don't have any examples of event.action ```system.email.password_reset.sent_message``` or ```system.email.account_unlock.sent_message``` currently.

Reply:
Thanks for the detailed reply!